File: leetcode/rust-leetcode/src/145.binary-tree-postorder-traversal.rs
// Definition for a binary tree node.
// #[derive(Debug, PartialEq, Eq)]
// pub struct TreeNode {
//   pub val: i32,
//   pub left: Option<Rc<RefCell<TreeNode>>>,
//   pub right: Option<Rc<RefCell<TreeNode>>>,
// }
// 
// impl TreeNode {
//   #[inline]
//   pub fn new(val: i32) -> Self {
//     TreeNode {
//       val,
//       left: None,
//       right: None
//     }
//   }
// }
use std::rc::Rc;
use std::cell::RefCell;
impl Solution {
    pub fn postorder_traversal(root: Option<Rc<RefCell<TreeNode>>>) -> Vec<i32> {
        fn dfs(root: Option<Rc<RefCell<TreeNode>>>, v: &mut Vec<i32>) {
            match root {
                None => {},
                Some(node) => {
                    let node = node.borrow();
                    dfs(node.left.clone(), v);
                    dfs(node.right.clone(), v);
                    v.push(node.val);
                }
            }
        }
        let mut ans = Vec::new();
        dfs(root, &mut ans);
        ans
    }
}

File: leetcode/python/145.binary-tree-postorder-traversal.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def postorderTraversal(self, root: Optional[TreeNode]) -> List[int]:
        def postorder(node: Optional[TreeNode], post: List[int]):
            if node is None:
                return
            postorder(node.left, post)
            postorder(node.right, post)
            post.append(node.val)
        ans = []
        postorder(root, ans)
        return ans

